src/commands/playground/util.rs

```diff
@@ -198,26 +197,44 @@ pub fn hoise_crate_attributes(code: &str, after_crate_attrs: &str, after_code: &
 /// Utility used by the commands to wrap the given code in a `fn main` if not already wrapped.
 /// To check, whether a wrap was done, check if the return type is Cow::Borrowed vs Cow::Owned
 /// If a wrap was done, also hoists crate attributes to the top so they keep working
-pub fn maybe_wrap(code: &str, result_handling: ResultHandling) -> Cow<'_, str> {
-	if code.contains("fn main") || code.contains("#![no_main]") {
-		return Cow::Borrowed(code);
-	}
+pub fn maybe_wrap(code: &str, result_handling: ResultHandling) -> String {
+	use quote::quote;
+	use syn::{parse::Parse, *};
 
-	// fn main boilerplate
-	let after_crate_attrs = match result_handling {
-		ResultHandling::None => "fn main() {\n",
-		ResultHandling::Discard => "fn main() { let _ = {\n",
-		ResultHandling::Print => "fn main() { println!(\"{:?}\", {\n",
-	};
+	struct Inline {
+		attrs: Vec<Attribute>,
+		exprs: Vec<Stmt>,
+	}
 
-	// fn main boilerplate counterpart
-	let after_code = match result_handling {
-		ResultHandling::None => "}",
-		ResultHandling::Discard => "}; }",
-		ResultHandling::Print => "}); }",
-	};
+	impl Parse for Inline {
+		fn parse(input: parse::ParseStream) -> Result<Self> {
+			let attrs = Attribute::parse_inner(input)?;
+			let exprs = Block::parse_within(input)?;
+			for stmt in &exprs {
```

Comment:
Bonus to avoid the `clone`:

```suggestion
			for stmt in &mut stmts {
```